src/components/sets/io.test.js

```diff
@@ -0,0 +1,138 @@
+
+describe('io.js', () => {
+  describe('importing sets', () => {
+    it('can import sets from a JSON file', (done) => {
+      const setsTree = {
+        import: (processedJSON) => {
+          expect(processedJSON.length).toEqual(correctProcessedJSON.length);
+          // eslint-disable-next-line no-plusplus
+          for (let i = 0; i < 6; i++) {
+            expect(processedJSON[i].key).toEqual(correctProcessedJSON[i].key);
+            expect(processedJSON[i].name).toEqual(correctProcessedJSON[i].name);
+            expect(processedJSON[i].color).toEqual(correctProcessedJSON[i].color);
+            expect(processedJSON[i].set).toEqual(correctProcessedJSON[i].set);
+          }
+          done();
+        },
+      };
+      handleImportJSON({
+        datasetId: 'linnarsson-2018', setsType: 'cell', onError: () => {}, setsTree,
+      }, jsonString);
+    });
+
+    it('can import sets from a TSV file', (done) => {
+      const setsTree = {
+        import: (processedJSON) => {
+          expect(processedJSON.length).toEqual(correctProcessedJSON.length);
+          // eslint-disable-next-line no-plusplus
+          for (let i = 0; i < 6; i++) {
+            expect(processedJSON[i].key).toEqual(correctProcessedJSON[i].key);
+            expect(processedJSON[i].name).toEqual(correctProcessedJSON[i].name);
+            expect(processedJSON[i].color).toEqual(correctProcessedJSON[i].color);
+            if (correctProcessedJSON.set) {
+              expect(processedJSON[i].set).toEqual(correctProcessedJSON[i].set);
+            }
+          }
+          done();
+        },
+      };
+      handleImportTabular({
+        setsTree,
+      }, tabularString);
+    });
+  });
+
+  describe('exporting sets', () => {
+    let factorsTree;
+    beforeEach(() => {
+      factorsTree = new SetsTree();
+      factorsTree.appendChild(new SetsTreeNode({
+        setKey: 'another-selection',
+        name: 'Another selection',
+        color: [128, 128, 128],
+        set: ['1972', '4047'],
```

Comment:
I prefer fixtures to be as simple as possible: Filling it in with 0s or 'foo' and 'bar' makes it clear that there's nothing in the data that really matters, and avoids distractions when reading.